node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/v2/filter/NodeIdentifier.java

```diff
@@ -31,9 +37,23 @@
     private final Zone zone;
     private final NodeRepository nodeRepository;
 
+    private final AtomicReference<List<Node>> nodes;
+    private static final Logger logger = Logger.getLogger(NodeIdentifier.class.getName());
+
     NodeIdentifier(Zone zone, NodeRepository nodeRepository) {
         this.zone = zone;
         this.nodeRepository = nodeRepository;
+        nodes = new AtomicReference<>(nodeRepository.getNodes());
+        new ScheduledThreadPoolExecutor(1, ThreadFactoryFactory.getDaemonThreadFactory("node-identifier-refresh"))
+                .scheduleAtFixedRate(this::updateNodeRepoCache, 1, 1, TimeUnit.MINUTES);
```

Comment:
Don't you want initialDelay in scheduleAtFixedRate() to be 0?

Reply:
No, It's already initialized above.
An alternative approach would be to set 0 , but then we need to ensure that the value is indeed set when using it.

Reply:
Sorry,  yes, this is good